### PR TITLE
add new check between the xml and zip version for local upgrade

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | add new check between the xml and zip version for local upgrade
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| Sponsor company   | -
| How to test?      | Use XML in another version. See alert

### CODE REVIEW

File: classes/TaskRunner/Miscellaneous/UpdateConfig.php [hunk 1]
```diff
@@ -95,12 +95,23 @@ public function run()
...
+            if (!empty($xmlFile) && !file_exists($fullXmlPath)) {
                 $this->error = true;
                 $this->logger->info($this->translator->trans('File %s does not exist. Unable to select that channel.', [$xmlFile]));
 
                 return false;
             }
+
+            $xmlVersion = $this->getXmlVersion($fullXmlPath);
+
+            if ($xmlVersion != $targetVersion) {
```
Comment (Quetzacoalt91) [inline classes/TaskRunner/Miscellaneous/UpdateConfig.php:108]:
```suggestion
            if ($xmlVersion !== $targetVersion) {
```

**Review by ga-devfront — APPROVED** [review]
Same comment then @Quetzacoalt91 but for the rest I think everything is okay !

File: classes/TaskRunner/Miscellaneous/UpdateConfig.php (hunk 1 above)
Comment (M0rgan01) [reply classes/TaskRunner/Miscellaneous/UpdateConfig.php:108]: extractPrestashopVersionFromZip return a string, same for getXmlVersion.

Comment (Quetzacoalt91) [reply classes/TaskRunner/Miscellaneous/UpdateConfig.php:108]: Than having the same type is a solid reason to have a strict check, and disallow any type casting.

**Review by Quetzacoalt91 — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
Hi @M0rgan01 

Thank you for your PR, I tested it and it seems to works as you can see : 

[recording(241).webm](https://github.com/PrestaShop/autoupgrade/assets/100699445/e943de67-422b-4fde-9515-cf9a272d245c)

Tested from : 
1.7.8.11 to 8.1.6
8.0.5 to 8.1.6
8.1.5 to 8.1.6
8.1.6 to 9.0.0
8.1.5 to 9.0.0
8.0.5 to 9.0.0

Because the PR seems to works as expected, It's QA :heavy_check_mark: 

Thank you

**Pull request merged**